qp_qiime2/qp_qiime2.py

```diff
@@ -355,7 +355,13 @@ def call_qiime2(qclient, job_id, parameters, out_dir):
             metadata_fp = join(out_dir, 'metadata.txt')
             metadata.to_csv(metadata_fp, index_label='#SampleID', na_rep='',
                             sep='\t', encoding='utf-8')
-            q2Metadata = qiime2.Metadata.load(metadata_fp)
+            # in 2022.8.3 qiime2 has a bug and the current solution is to load
+            # the file twice; the plan is that in the future this will not be
+            # needed
+            metadata_columns = qiime2.Metadata.load(metadata_fp).columns
+            q2Metadata = qiime2.Metadata.load(
```

Comment:
I think it's worth noting that the QIIME2 developers have sanctioned this usage:
https://forum.qiime2.org/t/q2-metadata-how-to-use-default-missing-scheme-and-column-missing-schemes/24255/7

I believe this won't immediately break once QIIME2 is fixed in the next release. We will need to remember to remove it, however.